RiotSwiftUI/Modules/UserSessions/UserSessionsOverview/View/UserSessionsOverview.swift

```diff
@@ -68,6 +39,69 @@ struct UserSessionsOverview: View {
             viewModel.send(viewAction: .viewAppeared)
         }
     }
+    
+    @ViewBuilder
+    private var securityRecommendationsSection: some View {
+        if hasSecurityRecommendations {
+            SwiftUI.Section {
+                if !viewModel.viewState.unverifiedSessionsViewData.isEmpty {
```

Comment:
Instead of doing logic inside the view itself (because the view should be as dumb as possible), I would move it to the model. 
As a suggestion, you could create something like this:
```
enum UserSessionsOverviewSection {
    case unverifiedSession(Int)
    case currentSession(UserSessionCardViewData)
    case otherSessions([UserSessionListItemViewData])
}
```
And inside `viewState` (`UserSessionsOverviewViewState`) you could have just one array of sections [UserSessionsOverviewSection]. The view would iterate through the array of sections and create sections without any logic inside the view.

If you want you could take it even further, by not using enum but by using an array of sections that would confirm to a section protocol, that could return a needed view for a section. In this case you would follow SOLID open close principle, that is any additional section wouldn't require a change in the main view.

Reply:
Talked about if offline, agreed to clean it up in a future PR by introducing a Section enum and maybe extracting some views out into separate files